src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettingsProvider.cs

```diff
@@ -815,6 +816,47 @@ public Task RemoveProfileAsync(string profileName)
             });
         }
 
+        public Task UpdateProfileAsync(string profileName, Action<IWritableLaunchProfile> updateAction)
+        {
+            // Updates need to be sequenced
+            return _sequentialTaskQueue.ExecuteTask(async () =>
+            {
+                ILaunchSettings currentSettings = await GetSnapshotThrowIfErrors();
+                ILaunchProfile? existingProfile = null;
+                int insertionIndex = 0;
+                foreach (ILaunchProfile p in currentSettings.Profiles)
+                {
+                    if (LaunchProfile.IsSameProfileName(p.Name, profileName))
+                    {
+                        existingProfile = p;
+                        break;
+                    }
+                    insertionIndex++;
+                }
+
+                if (existingProfile is null)
+                {
+                    return;
+                }
```

Comment:
The caller might reasonably want to know that this failed, either via an exception or a return value.

Reply:
Hmm. We could return a `bool` to indicate if the change was successfully applied. I'm not sure about the exception--I don't think it is exceptional that the profile might be renamed or deleted before the update is applied.

If this fails, what is the caller going to do?

Reply:
> If this fails, what is the caller going to do?

Display a message to the user rather than fail silently. I can't think of anything better than that.

Reply:
I've updated the method to return a `bool`.